frontend/ui/elements/screensaver_menu.lua

```diff
@@ -0,0 +1,190 @@
+local Screensaver = require("ui/screensaver")
+local _ = require("gettext")
+
+local function screensaverType() return G_reader_settings:readSetting("screensaver_type") end
+local function screensaverDelay() return G_reader_settings:readSetting("screensaver_delay") end
+local function lastFile() return G_reader_settings:readSetting("lastfile") end
+
+return {
+    {
+        text = _("Use last book's cover as screensaver"),
+        enabled_func = function() return lastFile() ~= nil end,
+        checked_func = function()
+            if screensaverType() == "cover" then
+                return true
+            else
+                return false
+            end
+        end,
+        callback = function()
+            G_reader_settings:saveSetting("screensaver_type", "cover")
+        end
+    },
+    {
+        text = _("Use book status as screensaver"),
+        enabled_func = function() return lastFile() ~= nil end,
+        checked_func = function()
+            if screensaverType() == "bookstatus" then
+                return true
+            else
+                return false
+            end
+        end,
+        callback = function()
+            G_reader_settings:saveSetting("screensaver_type", "bookstatus")
+        end
+    },
+    {
+        text = _("Use random image from folder as screensaver"),
+        checked_func = function()
+            if screensaverType() == "random_image" then
+                return true
+            else
+                return false
+            end
+        end,
+        callback = function()
+            G_reader_settings:saveSetting("screensaver_type", "random_image")
+        end
+    },
+    {
+        text = _("Use reading progress as screensaver"),
+        enabled_func = function() return Screensaver.getReaderProgress ~= nil and lastFile() ~= nil end,
+        checked_func = function()
+            if screensaverType() == "readingprogress" then
+                return true
+            else
+                return false
+            end
+        end,
+        callback = function()
+            G_reader_settings:saveSetting("screensaver_type", "readingprogress")
+        end
+    },
+    {
+        text = _("Use message as screensaver"),
+        checked_func = function()
+            if screensaverType() == "message" then
+                return true
+            else
+                return false
+            end
+        end,
+        callback = function()
+            G_reader_settings:saveSetting("screensaver_type", "message")
+        end
+    },
+    {
+        text = _("Disable screensaver"),
```

Comment:
May be this could be named something like `Disabled (let screen/page as it is)`, just to not be considered "suspend disabled" (dunno why, as I'm quite used to this, but it got me perplexed when I tried it, waiting for something to happen...)